packages/dds/tree/docs/storage/compatibility.md

```diff
@@ -0,0 +1,273 @@
+# Compatibility
+
+This document provides concrete guidelines and strategies for organizing code that impacts SharedTree's persisted format.
+It then pivots to the current state of SharedTree's persisted format, and outlines steps to take to align the implementation
+with those guidelines.
+
+## Prerequisites
+
+[This document](../../../SchemaVersioning.md) provides general "best practices" for working with persisted data within the Fluid Framework.
+It's strongly recommended to read through and understand its rationale before continuing with this document,
+as most of the concrete recommendations presented henceforth fall out of those best practices.
+
+## What State is Persisted?
+
+A DDS's persisted format encompasses the format it uses for its summaries as well as its ops (due to [trailing ops](../../../README.md))
+including transitively referenced structured blob data.
+
+Since documents are stored outside of Fluid control (i.e. no type of central data migration is possible),
+DDSes necessarily commit to backwards compatibility of their format for all time.
+
+## Format Management
+
+The actively used format is stored and managed by an `IPersistedConfigStore`.
+This configuration dictates all pieces of policy which impact the set of allowed ops, write format, etc.
+
+This config store is serialized at summarization time,
+which ensures all SharedTree summaries contain sufficient format version information.
```

Comment:
What is the process of switching V1 -> V2 when deploying forward compatible changes? If summary needs to be committed to do V1-> V2 to change policy, how actively running clients learn about it?

Reply:
the proposal here doesn't involve summary commit. It's elaborated on in #14632. I'm assuming that idea comes from Matt's recent work on external migration APIs. Here we have a much more scoped problem, whereby a DDS wants to change something about its persisted format. It's an easier problem to solve due to better transactionality guarantees--all changes are within a single DDS and the implementers of any 'migration code' necessary have access to low-level APIs. Some of the principles here probably also work for cases like "I work on the ContainerRuntime and I want to change the ISummaryTree format," but I'm not aiming that big in any near future (IMO, best to try it out in tree2 first and evaluate from there).